core/trino-parser/src/main/java/io/trino/sql/tree/MatchNumberFunction.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+
+public class MatchNumberFunction
```

Comment:
This class is part of the IR, so instead of introducing a new node, we should consider using a synthetic function call node to represent it instead of introducing more AST nodes that are not part of the grammar.

Reply:
I removed `MatchNumberFunction`, `ClassifierFunction` and `PatternNavigationFunction`. Instead, I keep them in the form of `FunctionCall` until `LogicalIndexExtractor`, where they are processed and removed.

I thought I would similarly remove `LabelDereference` node, which is also a temporary node.
However, there's an issue: `LabelDereference` comes in a form of `DereferenceExpresion`. The `TranslationMap` rewrites the target column to a `Symbol`. In order to wrap it back into `DereferenceExpression`, I should create an `Identifier` out of the `Symbol`. It might fail, as the Symbol name is not necessarily a good Identifier name.

So far, I left `LabelDereference` as an AST node.

What do you think?